client/my-sites/email/email-forwarding/email-forwarding-add-new-compact.jsx

```diff
@@ -37,12 +39,22 @@ class EmailForwardingAddNewCompact extends Component {
 		} );
 	}
 
+	componentDidMount() {
+		this.isMounted = true;
+	}
+
+	componentWillUnmount() {
+		this.isMounted = false;
+	}
+
 	getInitialFields() {
 		return this.props.fields;
 	}
 
 	setFormState = ( fields ) => {
-		this.setState( { fields } );
+		if ( this.isMounted ) {
+			this.setState( { fields } );
```

Comment:
This is not particularly elegant IMO, but it prevents an issue where `formState.Controller` might call `setFormState` after the component has unmounted (causing React to warn about "potential memory leaks"). Why does that happen? Because `formState.Controller` will automatically debounce field validations. This means that the validation could finish after the form has been submitted (and this component has unmounted).

It might be a good idea to revisit this form validation at some point. Ideally I would like clearer separation of responsibility between this component and the parent component (where the actual `<form>` lives) when it comes to validation.